core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/CryptoUtil.java

```diff
@@ -231,20 +222,20 @@ public String encryptAndBase64Encode(byte[] plainText) throws
                 throw new CryptoException("A crypto service implementation has not been registered.");
             }
 
-            // Set the default crypto algorithm to be used when a cipher transformation is not found.
-            String algorithm = DEFAULT_CRYPTO_ALGORITHM;
+            String algorithm = null;
 
             String cipherTransformation = System.getProperty(CIPHER_TRANSFORMATION_SYSTEM_PROPERTY);
 
             if (cipherTransformation != null) {
-                CipherHolder cipherHolder = cipherTextToCipherHolder(cipherTextBytes);
-                if (cipherHolder != null) {
+                CipherMetaDataHolder
+                        cipherMetaDataHolder = cipherTextToCipherMetaDataHolder(cipherTextBytes);
+                if (cipherMetaDataHolder != null) {
```

Comment:
Did we verify this will not have an impact on previously encrypted data for asymmetric flow

Reply:
I think, during migration effort we can handle this. When there are previously encrypted data on asymmetric , we will have to do a migration anyway. This can also be handled from there i think.

Reply:
Verified this. No issues observed.